### PR TITLE
remove legacy, add strict, based by default.

### LINKED ISSUE #282: Re-add strict mode and add a `non-strict` option
The use case is that `legacy` sounds like it means it disables all based features (there's probably a use case for that). But a `non-strict` flag that sets strictness to `legacy` level but doesn't disable any based stuff would be good.

### LINKED ISSUE #296: `default_return` should default to true (do better)
# 🔥 Issues:
- It is less discover-able
- You have to enable it per module, so enabling it for a scripts folder is extremely painful
- It doesn't activate in gradually typed stubs/modules:
```py
def foo(): ...  # I wan't default return :(
```

# 🚒 Solution:
Default return is enabled by default for everything, except site-packages.

- instant discover-ablility
- you don't have to enable it per module, all you will ever have to do is disable it on edge case things like vendored files.

It will be enabled even on dynamic functions, but the functions will still be considered dynamic (for func body checking purposes).

This means that:
```py
def foo(): ...
```
**Will** be not be considered dynamic.

- if you still want dynamic return then type it with `Untyped` or `Any`
- This might not be up everyone's tree, so I think we should have three options for this setting, on/off and _something_ (where it favors explicit annotation in ambiguous cases)

### LINKED ISSUE #465: rename 'nested generics' to 'generic bounds' (or something)

### PR DESCRIPTION
**Pull request by KotlinIsland**
fix all the sus default config

- resolves #282
- resolves #296
- resolves #465

### CODE REVIEW

**Comment by KotlinIsland** [comment]
@DetachHead Wanna review this? it's mostly just renaming `--legacy` to `--non-strict`.

**Comment by DetachHead** [comment]
i think its bad enough that all the flags are implemented in triple negatives, idk if renaming `strict` to `non-strict` really helps make it less confusing

whats the reason behind changing this again?

**Comment by DetachHead** [comment]
https://github.com/KotlinIsland/basedmypy/issues/282#issuecomment-1236611259

File: test-data/unit/cmdline-based-baseline.test
Comment (DetachHead) [inline test-data/unit/cmdline-based-baseline.test:1]: were these formatting changes intentional

**Review by DetachHead — COMMENTED** [review]
change it to just `strict` and `no-strict` to match upstream

File: test-data/unit/cmdline-based-baseline.test
Comment (KotlinIsland) [reply test-data/unit/cmdline-based-baseline.test:1]: yeah

**Comment by KotlinIsland** [comment]
Failed CI is all upstreams fault

**Review by DetachHead — APPROVED** [review]

**Pull request merged**